src/lib/components/DarkModeControl.svelte

```diff
@@ -28,18 +32,9 @@
 
 <button type="button" role="switch" aria-checked={darkMode} class={cssClass} on:click={setColorMode}>
     {#if darkMode}
-    <!-- sun-fill icon -->
-    <svg aria-hidden="true" xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-sun-fill" viewBox="0 0 16 16">
-        <path d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"/>
-    </svg>
-        <span class="sr-only">Light</span>
+        <MoonStarsFill aria-hidden="true"/><span class="sr-only">Dark mode</span>
```

Comment:
why show the dark mode icon when you're in dark mode? 🤔 
The button changes the mode: "i want to switch to [sun (light)/ moon (dark)]" 

If reversing the images, the helper sr-only text should still tell the user what the action of the button is, when in dark mode clicking it would mean the action is changing to light mode (need to reverse the text in that case)

Reply:
@VeckoTheGecko pushed some changes but not sure on this one 🙃

Reply:
Thanks for the pushes and attention to detail @dmlb ! Wrt. this, I think in our context where we're using a sun/moon to show light/dark mode, I have only ever seen this done by displaying "the icon for the current theme being shown" and not "the icon for the theme to switch to". For example, the [Pydata theme](https://pydata-sphinx-theme.readthedocs.io/en/stable/) (used for doumentation for various Python projects like numpy, pandas, etc). Similarly (though slightly different) are switches that have sun/moon icons included (e.g. [`sveltekit-dark-mode` theme](https://github.com/CaptainCodeman/sveltekit-dark-mode). I vaguely recall other instances, but I cant think of them at the moment.

I find the proposed implementation intuitive/familiar, but that may just be me. An alternative would be to use `toggle-on` and `toggle-off` switch icons like in [Wes McKinneys book website](https://wesmckinney.com/book/) (which I think is using JupyterBook), or implement something else.

Also, I'm wondering if the `sr-only` tags ffor light/dark mode might be confusing at the moment, and maybe we should go for a single `<span class="sr-only">Toggle dark mode</span>` option, which I've seen in the `aria-label` of some websites (1000% not my area of expertise though, so deferring to your experience here 😄 )

Reply:
yeah, just aria-hidden icons then :D